cas-server-core-api-services/src/main/java/org/jasig/cas/services/RegisteredServiceAuthenticationPolicy.java

```diff
@@ -0,0 +1,44 @@
+package org.jasig.cas.services;
+
+import java.io.Serializable;
+import java.util.Set;
+
+/**
+ * This is {@link RegisteredServiceAuthenticationPolicy} that describes how a service
+ * should handle authentication requests.
+ *
+ * @author Misagh Moayyed
+ * @since 4.3.0
+ */
+public interface RegisteredServiceAuthenticationPolicy extends Serializable {
```

Comment:
There is another `AuthenticationPolicy` interface that is used within `PolicyBasedAuthenticationManager`. Whenever you were going over this interface earlier my mind jumped to that similarly named interface.

Maybe we should rename this to help differentiate the 2 interfaces better? This currently seems very MFA centric (obviously), do you see this being extended down the line with non-MFA use cases? or should we just rename it to something like `RegisteredServiceMultifactorPolicy`?